### DISCUSSION TITLE
QRControl command for Hero 10 Black to implement conditional Live Streaming

### DISCUSSION THREAD

**Discussion by AM-Novus** [gopro/labs#237] [Q&A]
Trying to write a QRControl command for Hero 10 black to Stream when motion detection is higher than 10%, I have been able to implement these features independently but have had no luck combining them, is this even possible?

**Comment by dnewman-gpsw** [comment]
It is not fully possible on HERO10, it will be easier in the next HERO11 Labs release.

The only hack for HERO10 would be to use a command like  !SM4-9D2!G

It will detect motion, start a normal motion capture (when motion is detected), then immediate stop, then start a live stream.

**Reply by AM-Novus** [reply]
What about Actions and Conditions? Why is a command like ">m5!GM" which implies if more than 5% of the pixels are moving, start the live stream not responding? I tried this command, the QR code is understood successfully but the camera doesn't implement the commands

**Reply by dnewman-gpsw** [reply]
That is an upcoming command for HERO11. From "Conditionals Based on Camera Status" :  "Coming soon (HERO11), new conditional commands. "

HERO11 official support might be out around March.  In testing now.

Weirdly there is an un-tested, un-official HERO10 build made with some these commands implemented. It might never get an official release, but you can try it.  https://www.dropbox.com/s/hspp75b3yx533b9/HERO10LabsV15071.zip?dl=1

**Reply by dnewman-gpsw** [reply]
I just tested this, and it is not likely to work.  This particular feature wasn't designed for HERO10, and still needs work for HERO11.  Just not ready yet.

**Reply by AM-Novus** [reply]
Really appreciate your responses, I will try to see if I can get hold of a HERO11 for now, just wanted to know if you have any idea on if there are any plans to add conditional and actions compatibility to HERO 10 in any future releases?

**Reply by dnewman-gpsw** [reply]
HERO11 support coming in March (ish.) Nothing official on the roadmap for HERO10, although I'm also testing H10, just in case that can change.

**Reply by dnewman-gpsw** [reply]
I fixed HERO10 motion trigger to more more like H11 (H10 is half the system memory, so it is not exactly the same, but close.) @AM-Novus please be my guineapig. Updated Firmware for HERO10 https://www.dropbox.com/s/xe7zwjkb6xr2ggh/HERO10LabsV15071a.zip?dl=1

Start with: oMDSPL=10"motion $m"!R   (this will just print the LCD the motion detected.) The will give you feedback to the sensitivity, to determine the best threshold. 

Then you can run a command like:  <r0>m5!S>r0<m.2!E!R

as pseudo code:
```
loop:
  if(recording < 0 && motion > 5%) StartCapture();
  if(recording > 0 && motion < 0.2%) EndCapture();
  goto loop;
```

Code has to do a little more to emulate hold times: <r0>m5!S+=A0>r0<m.2=A+1~=A0>A10!E!R

as pseudo code:
```
loop:
  if(recording < 0 && motion > 5%) StartCapture(), A=0;
  if(recording > 0 && motion < 0.2%) A=A+1; else A=0;
  if(A > 10) EndCapture();
  goto loop;
```

An auto loop Live stream motion trigger could be something like: <l1r1+!W+!2N<r0>m5!GLC+!15N+=A0>r0<m.2=A+1~=A0>A20!E!R

as pseudo code:
```
if(loop<1)  SetResoltion(1080), JoinWiFi(), Sleep(2seconds);  
loop:
  if(recording < 0 && motion > 5%) GoLive(),Sleep(15seconds), A=0;
  if(recording > 0 && motion < 0.2%) A=A+1; else A=0;
  if(A > 20) EndCapture();
  goto loop;
```

When testing all of the above, I found the !GL (with out the C) didn't work.  Seems the recording flag needs a file to captured.  
Currently there is no cancel feature, so either pull the battery or scan different QR Code.

**Comment by AM-Novus** [comment]
Thank you so much for getting back so quickly, the command does part of want I would like to have. I am trying to setup my gopro such that it starts livestreaming in seconds of a motion being captured and then once it thinks there isn't enough motion on the screen it should be able to stop stream and then repeat the command. Is there a way I can help out with this feature or something along that lines, I would love for this feature to be there on GoPro Black 10. Thanks
